mllib/src/test/scala/org/apache/spark/ml/feature/Word2VecSuite.scala

```diff
@@ -134,13 +134,20 @@ class Word2VecSuite extends SparkFunSuite with MLlibTestSparkContext with Defaul
       .fit(docDF)
 
     val expectedSimilarity = Array(0.2608488929093532, -0.8271274846926078)
-    val (synonyms, similarity) = model.findSynonyms("a", 2).rdd.map {
+    val result = model.findSynonyms("a", 2).rdd.map {
       case Row(w: String, sim: Double) => (w, sim)
-    }.collect().unzip
+    }.collect()
+    val (synonyms, similarity) = result.unzip
 
     assert(synonyms === Array("b", "c"))
     expectedSimilarity.zip(similarity).foreach {
-      case (expected, actual) => assert(math.abs((expected - actual) / expected) < 1E-5)
+      case (expected, actual) => assert(expected ~== actual absTol 1E-5)
+    }
+
+    result.zip(model.findSynonymsArray("a", 2)).foreach {
```

Comment:
Technically, collect() may not return elements in the same order each time, so this will be more robust if we convert each result to a map: synonym -> similarity and compare the maps.  Could you please do that?